File: dataset_reduzido/codenet/Python/Code/s752993784.py
s = list(input())
mods = [0]*(len(s))
countRemainder = [0] * 2019
cnt = 0


m = 1
mod = 0
for i in range(len(s)):
    mod += int(s[len(s) - i - 1]) * m

    mod %= 2019
    m *= 10
    m %= 2019

    countRemainder[mod] += 1

countRemainder[0] += 1

for i in range(2019):
    cnt += int(countRemainder[i] * (countRemainder[i] - 1) / 2)

print(cnt)

File: output/LLaMa/codenet/Python/Java/s752993784.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String s = sc.next();
        int[] mods = new int[s.length()];
        int[] countRemainder = new int[2019];
        int cnt = 0;

        long m = 1;
        int mod = 0;

        for (int i = s.length() - 1; i >= 0; i--) {
            mod = (int) ((mod + (s.charAt(i) - '0') * m) % 2019);
            m = (m * 10) % 2019;

            countRemainder[mod]++;
        }

        countRemainder[0]++;

        for (int i = 0; i < 2019; i++) {
            cnt += (long) countRemainder[i] * (countRemainder[i] - 1) / 2;
        }

        System.out.println(cnt);
    }
}